libpermian/plugins/anaconda_webui/__init__.py

```diff
@@ -0,0 +1,443 @@
+            for repo in self.additional_repos:
+                kernel_cmdline += f' inst.addrepo={self.installation_source.repo[repo][self.architecture]}'
+
+        if self.kernel_cmdline:
+            kernel_cmdline += f' {self.kernel_cmdline}'
+
+        # Assemble virt-install command
+        cmd = ['virt-install', '--connect', self.hypervisor_host, '--noautoconsole',
+            '-n', self.vm_name, '--os-variant', 'rhel-unknown', '--location', location,
+            '--memory', '4096', '--vcpus', '2', '--disk', 'size=10',
+            '--extra-args', kernel_cmdline]
+
+        LOGGER.info('Running: ' + repr(cmd))
+        process = subprocess.run(cmd ,check=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+        self.log(process.stdout.decode())
+
+        # Wait for machine to start and get IP
+        self.log('Waiting for IP', show=True)
+        startup_timeout = 120 + time.time()
+        while True:
+            if startup_timeout < time.time():
+                Exception('Timeout, VM still doesn\'t have IP')
+            time.sleep(10)
+            output = self._virsh_call(['domifaddr', self.vm_name])
+            if output:
+                break
+
+        return output.split()[3].split('/')[0]
+
+    def _virsh_call(self, args, check=False):
+        """ Runs virsh with specified arguments
+
+        :param args: virsh arguments
+        :type args: list
+        :param check: Check if command was successful, defaults to False
+        :type check: bool, optional
+        :return: CompletedProcess
+        :rtype: subprocess.CompletedProcess
+        """
+        cmd = ['virsh', '-q', '--connect', self.hypervisor_host] + args
+        LOGGER.debug('Running: ' + repr(cmd))
+        return subprocess.run(cmd, check=check, stdout=subprocess.PIPE).stdout.decode()
+
+    def _clone_repo(self, repo, branch, dir):
+        """ Clones git repository
+
+        :param repo: Repository URL
+        :type repo: str
+        :param branch: Branch name
+        :type branch: str
+        :param dir: Directory where the repo should be cloned
+        :type dir: str
+        """
+        git_cmd = ['git', 'clone', '--depth', '1', '-b', branch, repo, dir]
+        LOGGER.debug('Running: ' + repr(git_cmd))
+        subprocess.check_call(git_cmd)
+
+    def _get_test_repo_dir(self):
+        """ Clones test repository with the same branch used for anaonda repo, if test repo doesn't already exist """
+        self.test_repo_dir = os.path.join(self.temp_dir, self.test_repo_name)
+        if  not os.path.isdir(self.test_repo_dir):
+            self.log(f'Clonning test repo {self.test_repo_url}')
+            self._clone_repo(self.test_repo_url, self.git_anaconda_branch, self.test_repo_dir)
+
+    def _clone_common(self):
+        """ Clones repositories common to all Anaconda WebUI tests """
+        self.log(f'Clonning common repositories')
+        temp_cockpit = os.path.join(self.temp_dir, 'temp_cockpit')
```

Comment:
This sounds like a misuse of the temp directory as if I understand it correctly, this method does only use it to temporarily create a `temp_cockpit` directory there which it then removes effectively using it as any temporary directory. Instead of using the `self.temp_dir` for this purpose here, I'd suggest using https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory which can be used inside the with statement ensuring that whatever remains in the directory is cleaned up when leaving the context.